### DISCUSSION TITLE
Long running async task

### DISCUSSION THREAD

**Discussion by MTRNord** [hecrj/iced#954] [Q&A]
Hi I have an API that requires polling to get updates. I am aware about `Command::perform` for short term async execution. How could I however integrate with never exiting async execution? 

This specifically would be https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/struct.Client.html#method.sync for my case or https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/struct.Client.html#method.sync_with_callback

the first one emits it data to a struct implementing a specific trait or the second one is just a callback which actually is a loop internally.

So I need either of these hooked in a way where I can trigger the UI to update. Any ideas on how to archive this?

I have also seen Subscriptions but I am honestly not sure how to use them or if they work with this at all :) I can certainly add my Messagges I want to emit to a list and from that maybe for am a Subscription. But to get there I would need some help on how to do it correct/efficent in this case :)

**Comment by 13r0ck** [comment]
Does [this](https://github.com/iced-rs/cookbook/tree/main/0.3/render_to_framebuffer_and_save) help?

**Comment by reivilibre** [comment]
Thinking about this myself, I wonder if [`Subscription`](https://docs.rs/iced/0.3.0/iced/type.Subscription.html) might be suitable.

I notice the [stopwatch example](https://github.com/hecrj/iced/blob/a08e4ebccbb72f9cf6fca01047e0b46a482ca9ea/examples/stopwatch/src/main.rs) uses it to update as time elapses.

It seems like you can create your own subscription by implementing a [`Recipe`](https://docs.rs/iced_futures/0.3.0/iced_futures/subscription/trait.Recipe.html) and using `Subscription::from_recipe`...

If you get anywhere with this, let me know :D

**Reply by reivilibre** [reply]
The Recipe docstring also links to the stopwatch and download progress examples, so those two might help
